csrc/scheduler/tools/inlining.cpp

```diff
@@ -211,8 +228,34 @@ size_t MaxPosCalculator::getMaxProducerPosFromConsumer(
       }
 
       IterDomain* c_id = *consumer_it;
-      if (!inliningGraph().disjointValSets().strictAreMapped(p_id, c_id) ||
-          !isAllowedID(c_id, consumer, best_effort, true, false, true)) {
+
+      // We can inline past consumer IDs that are not connected to the producer.
+      //
+      // For example, an MmaOp with no broadcasts could contain the following:
+      //  tv0:
+      //    root/logical: [ iS0, iS1 ]
+      //    loop: [ iS0, bS7, iS1 ]
+      //  tv1:
+      //    root/logical: [ iS2, iS3 ]
+      //    loop: [ bS8, iS2, iS3 ]
+      //  tv2:
+      //    root/logical/loop: [ iS4, iS5, rS6 ]
+      //
+      //  iS4 maps to iS0 so when producer==tv0 we inline past iS0. When
+      //  producer==tv1, iS4 doesn't map to anything in tv1 and is not used for
+      //  indexing, and bS8 is a loop broadcast so we inline past the first ID
+      //  in that case also. Similarly, we inline past iS5, iS2, and bS7.
+      if ((loop_path_groups.empty() ||
+           loop_path_groups.count(inliningGraph().toGroup(p_id)) ||
+           loop_path_groups.count(inliningGraph().toGroup(c_id))) &&
```

Comment:
I think separating this logic would be easier to follow, like lines 222-224.

```suggestion
      // We can inline past consumer IDs that are not connected to the producer.
      //
      // For example, an MmaOp with no broadcasts could contain the following:
      if (loop_path_groups.has_value() &&
           (!loop_path_groups.count(inliningGraph().toGroup(p_id)) ||
           !loop_path_groups.count(inliningGraph().toGroup(c_id)))) {
           continue;
       }    
```